src/arkode/arkode_io.c

```diff
@@ -34,6 +35,347 @@
+                           ARKodeSetAdaptivityAdjustment,
+                           ARKodeSetSmallNumEFails,
+                           ARKodeSetMaxErrTestFails,
+                           ARKodeSetMaxConvFails};
+
+  int num_long_keys = 1;
+  const char* long_keys[] = {"arkode.max_num_steps"};
+  arkLongSetFn long_set[] = {ARKodeSetMaxNumSteps};
+
+  int num_real_keys = 18;
+  const char* real_keys[] = {"arkode.nonlin_crdown",
+                             "arkode.nonlin_rdiv",
+                             "arkode.delta_gamma_max",
+                             "arkode.nonlin_conv_coef",
+                             "arkode.init_step",
+                             "arkode.min_step",
+                             "arkode.max_step",
+                             "arkode.stop_time",
+                             "arkode.fixed_step",
+                             "arkode.step_direction",
+                             "arkode.cfl_fraction",
+                             "arkode.safety_factor",
+                             "arkode.error_bias",
+                             "arkode.max_growth",
+                             "arkode.min_reduction",
+                             "arkode.max_first_growth",
+                             "arkode.max_efail_growth",
+                             "arkode.max_cfail_growth"};
+  arkRealSetFn real_set[] = {ARKodeSetNonlinCRDown,
+                             ARKodeSetNonlinRDiv,
+                             ARKodeSetDeltaGammaMax,
+                             ARKodeSetNonlinConvCoef,
+                             ARKodeSetInitStep,
+                             ARKodeSetMinStep,
+                             ARKodeSetMaxStep,
+                             ARKodeSetStopTime,
+                             ARKodeSetFixedStep,
+                             ARKodeSetStepDirection,
+                             ARKodeSetCFLFraction,
+                             ARKodeSetSafetyFactor,
+                             ARKodeSetErrorBias,
+                             ARKodeSetMaxGrowth,
+                             ARKodeSetMinReduction,
+                             ARKodeSetMaxFirstGrowth,
+                             ARKodeSetMaxEFailGrowth,
+                             ARKodeSetMaxCFailGrowth};
+
+  int num_bool_keys = 4;
+  const char* bool_keys[] = {"arkode.nonlinear",
+                             "arkode.clear_stop_time",
+                             "arkode.no_inactive_root_warn",
+                             "arkode.reset_accumulated_error"};
+  arkBoolSetFn bool_set[] = {ARKodeSetNonlinear,
+                             ARKodeClearStopTime,
+                             ARKodeSetNoInactiveRootWarn,
+                             ARKodeResetAccumulatedError};
+
+  int i, j, retval;
+  for (i = 1; i < argc; i++)
+  {
+    sunbooleantype arg_used = SUNFALSE;
+
+    /* skip command-line arguments that do not begin with "arkode." */
+    if (strncmp(argv[i], "arkode.", 7) != 0) { continue; }
```

Comment:
Removes "magic constant"
```suggestion
    static const char *prefix = "arkode.";
    if (strncmp(argv[i], prefix, strlen(prefix)) != 0) { continue; }
```